config.go

```diff
@@ -153,11 +164,30 @@ func (c *ClientConfig) setDefaults(cfg *Config) {
 	c.Timeout = cfg.ClientTimeout
 }
 
-func (c *ClientConfig) validate() error {
+func (c *ClientConfig) validate(minLifetime time.Duration, logger *logrus.Entry) error {
 	if c.Key == "" {
 		return errors.New("No key in config")
 	}
 
+	keyPair, err := tls.LoadX509KeyPair(c.Cert, c.Key)
```

Comment:
`client.resolveKeyPair()` allows customizing the directory where `c.Cert` and `c.Key` are located; it also sets `c.Cert = c.Key` in the case where `c.Cert` isn't specified.  Trying to load the key pair here, before `resolveKeyPair` is called, is likely to fail in these two cases.

Reply:
This is an excellent find, I will fix this